### PR TITLE
Prefer `cloud_device_id` when matching connections

### PR DESCRIPTION
**Pull request by ahemphill**

### CODE REVIEW

File: tools/scoring/score/dimensions/entity_connection_identification.py [hunk 1]
```diff
@@ -30,23 +33,30 @@ class EntityConnectionIdentification(Dimension):
...
     """Distill individual connections from each entity
     prior to inclusion in sets for global comparison."""
+    Connection = namedtuple('Connection', ['target', 'connection'])
+
     return [
-        tup for tup in (((cloud_device_id, connection)
-                         for connection in entity.connections)
-                        for cloud_device_id, entity in file.items()
-                        if entity.connections is not None) for tup in tup
+        Connection(entity.code, connection) for entity in file.values()
```
Comment (charbull) [inline tools/scoring/score/dimensions/entity_connection_identification.py:42]: can you please write this with the normal syntax, it is hard to follow.

File: tools/scoring/score/dimensions/entity_connection_identification.py [hunk 2]
```diff
@@ -30,23 +33,30 @@ class EntityConnectionIdentification(Dimension):
...
     for easy comparison using intersection."""
-    return set([
-        f'{target} {connection.ctype} {connection.source}'
-        for target, connection in connections
-    ])
+    condensed = set()
+    for cn in connections:
+      cdid_or_code = lambda code_or_guid: next(
+          entity.cloud_device_id or entity.code for entity in file.values()
+          if code_or_guid in [entity.code, entity.guid])
```
Comment (charbull) [inline tools/scoring/score/dimensions/entity_connection_identification.py:54]: can you please avoid lambda, my brain gets heated

File: tools/scoring/score/dimensions/entity_identification.py [hunk 3]
```diff
@@ -32,24 +32,27 @@ class EntityIdentification(Dimension):
...
-  def _list_ids_virtual(self, file: DeserializedFile) -> List[CloudDeviceId]:
-    """Generates list of `cloud_device_id`s representing
-    reporting entities with canonical types that
-    are linked to by virtual entities."""
-    return [
-        cloud_device_id for source_list in (
-            (file[link.source].cloud_device_id for link in entity.links)
-            for entity in filter(self.is_entity_canonical,
-                                 filter(self.is_entity_virtual, file.values())))
+        cloud_device_id for source_list in ((file[link.source].cloud_device_id
```
Comment (charbull) [inline tools/scoring/score/dimensions/entity_identification.py:53]: simplify the syntax here too with an `old` for loop please

File: tools/scoring/tests/samples/solution/real_world_solution.yaml [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+          no_units: no-units
+    reactive_power_sensor:
+      present_value: data.kVAR.present-value
+      units:
+        key: data.kVAR.present-value
+        values:
+          kilovolt_amperes_reactive: kilo-volt-amperes-reactive
+  type: METERS/EM_PWM_AVCM_PLPM_PVCM
+  guid: a51ee956-0ea4-47ae-b398-b1b38eb103fa
+power-meter-EV_L1C:
```
Comment (charbull) [inline tools/scoring/tests/samples/solution/real_world_solution.yaml:231]: can you please leave empty lines between entities for visibility

File: tools/scoring/tests/samples/solution/real_world_solution.yaml [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+        values:
+          degrees_fahrenheit: degrees-fahrenheit
+  type: HVAC/WEATHER_OA
+  guid: 0899c249-3854-40d8-ba14-2dfe64e36bcb
+power-meter-ELEV:
+  cloud_device_id: 2566753989830144
+  connections:
+    US-MTV-1804:
+    - CONTAINS
+  id: CDM/2566753989830144
```
Comment (charbull) [inline tools/scoring/tests/samples/solution/real_world_solution.yaml:83]: those must be deleted from the test file as we deprecated them, soon the parser will throw errors.

**Review by charbull — CHANGES_REQUESTED** [review]

File: tools/scoring/score/dimensions/entity_connection_identification.py (hunk 2 above)
Comment (charbull) [reply tools/scoring/score/dimensions/entity_connection_identification.py:54]:
```
for cn in connections:
  foo(code_or_guid)
   
def foo(code_or_guid):
 for entity in file.values():
  if code_or_guid in [entity.code, entity.guid]
     return entity.cloud_device_id

**Review by charbull — CHANGES_REQUESTED** [review]

File: tools/scoring/score/dimensions/entity_connection_identification.py (hunk 1 above)
Comment (ahemphill) [reply tools/scoring/score/dimensions/entity_connection_identification.py]: https://github.com/google/digitalbuildings/pull/561/commits/9eec8f21541d609a54f52c3bb3207c4fd4aa5832

File: tools/scoring/score/dimensions/entity_identification.py (hunk 3 above)
Comment (ahemphill) [reply tools/scoring/score/dimensions/entity_identification.py:53]: https://github.com/google/digitalbuildings/pull/561/commits/9eec8f21541d609a54f52c3bb3207c4fd4aa5832

File: tools/scoring/tests/samples/solution/real_world_solution.yaml (hunk 4 above)
Comment (ahemphill) [reply tools/scoring/tests/samples/solution/real_world_solution.yaml:231]: https://github.com/google/digitalbuildings/pull/555/commits/ff694d3aaa11f680e25c31a36a3cc9a6c94230a3

File: tools/scoring/tests/samples/solution/real_world_solution.yaml (hunk 5 above)
Comment (ahemphill) [reply tools/scoring/tests/samples/solution/real_world_solution.yaml:83]: https://github.com/google/digitalbuildings/pull/555/commits/ff694d3aaa11f680e25c31a36a3cc9a6c94230a3

File: tools/scoring/score/dimensions/entity_connection_identification.py (hunk 2 above)
Comment (ahemphill) [reply tools/scoring/score/dimensions/entity_connection_identification.py]: https://github.com/google/digitalbuildings/pull/561/commits/9eec8f21541d609a54f52c3bb3207c4fd4aa5832

File: tools/scoring/score/dimensions/entity_identification.py (hunk 3 above)
Comment (ahemphill) [reply tools/scoring/score/dimensions/entity_identification.py:53]: https://github.com/google/digitalbuildings/pull/552/commits/b26745428813dce292c368603a7b003ba22318f9

**Review by charbull — APPROVED** [review]
